### PR TITLE
[CI] Add end-to-end V1 min_tokens test coverage

### LINKED ISSUE #21950: [Bug]: Verify that the `min_tokens` sampling parameter is working and covered by CI tests
It has been reported that this is not working in the latest V1 code. I'm fairly sure we have this in our CI but perhaps it was only for V0.

It may be related to the recent refactoring to move the implementation to a LogitsProcessor.

Ref: https://github.com/vllm-project/vllm/issues/21672#issuecomment-3127534960

cc @afeldman-nm

### PR DESCRIPTION
**Pull request by arjunbreddy22**
# Essential Elements of an Effective PR Description Checklist

- [X] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [X] The test plan, such as providing test command.
- [X] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.

## Purpose
Add comprehensive end-to-end CI coverage for V1 `min_tokens`.  
Resolves #21950 (verify and add CI coverage).

Includes bug-repro tests:
- Stop strings bypassing `min_tokens` (known bug #21987; being fixed in PR #22014).
- EOS token behavior with `min_tokens` (potential logits-processor bug; now appears fixed).

Scope:
- Tests only; no functional code changes.
- New file: `tests/v1/e2e/test_min_tokens.py`.

## Test Plan
- All tests target the **V1** engine with a small model (`facebook/opt-125m`) for fast CPU execution.  
- Deterministic setup via `temperature=0.0` and `enforce_eager=True`.

Run locally:
```bash
# From repo root
VLLM_USE_V1=1 python -m pytest tests/v1/e2e/test_min_tokens.py -v 
```

## Test Result
Latest run (CPU, ~26s):

- **7 passed**
- **3 xfailed** (stop-strings wide and simple stop lists)
- **3 xpassed** (EOS-related and a guaranteed early-trigger case)

Example summary:

- **PASSED**: baseline and edge cases.
- **XFAIL**: stop-strings bypass `min_tokens` (expected until fully fixed).
- **XPASS**: EOS-related tests and one stop-strings stress test indicate upstream fixes are effective in those paths.

Note:
- XFAILs use decorator marks with `strict=False`, so when upstream fixes land, they will XPASS without breaking CI.

## (Optional) Documentation Update
N/A

### CODE REVIEW

File: tests/v1/e2e/test_min_tokens.py [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+from typing import Optional, Union
+
+import pytest
+
+from vllm import LLM, SamplingParams
+from vllm.outputs import RequestOutput
+
+# Test configuration
+TEST_MODEL = "facebook/opt-125m"  # Small model for fast CI execution
+TEMPERATURE = 0.0  # Deterministic generation for consistent testing
```
Comment (njhill) [inline tests/v1/e2e/test_min_tokens.py]: suggest to rename this to `GREEDY`

File: tests/v1/e2e/test_min_tokens.py [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+    """
+    # If the bug is fixed upstream, this test will XPASS
+
+    sampling_params = SamplingParams(
+        min_tokens=25,
+        max_tokens=25,  # Force exact length
+        temperature=TEMPERATURE)
+
+    prompt = "The capital of France is"
+    outputs = llm_v1.generate([prompt], sampling_params)
```
Comment (njhill) [inline tests/v1/e2e/test_min_tokens.py]: @arjunbreddy22 could you make calls here without and with `min_tokens`, check in the "without" case that < 25 tokens are output with finish_reason of `stop` and stop_reason `None`. And also in the "with" case verify that none of the generated token ids == the eos token id.

**Review by njhill — COMMENTED** [review]
Thanks very much @arjunbreddy22!

So it looks like you verified that `min_tokens` appears to be working in non-stop-sequence case?

Please see my inline comment though, would be good to sanity check that it really is suppressing eos and not just generating 25 tokens anyhow.

**Comment by njhill** [comment]
cc  @vadimkantorov

**Comment by arjunbreddy22** [comment]
Thanks for the review! I’ve updated the test to run both with and without min_tokens, adding checks to ensure that in the without case fewer than 25 tokens are generated with finish_reason set to stop and no stop_reason, and in the with case no EOS token ID appears. I also renamed TEMPERATURE to GREEDY for clarity.

**Comment by arjunbreddy22** [comment]
On your earlier question by the way. Yes, min_tokens is working in the non stop sequence case. Non stop sequence tests pass when I ran it myself (basic_min_tokens_no_stop, min_tokens_zero, min_equals_max_no_stop, large_min_tokens, min_tokens_with_empty_stop_list). Also, for your third comment I've verified that in the "without" case (which I added), < 32 (I changed from 25 to 32) of the tokens are output and the finish_reason was stop. And in the "with" case none of the generated tokens == the eos token id.

**Review by njhill — APPROVED** [review]
Thanks a lot @arjunbreddy22!

**Comment by njhill** [comment]
@vadimkantorov any idea what's going on here? `min_tokens` appears to be working ... could it somehow have been fixed since 0.9.2?

**Comment by vadimkantorov** [comment]
Hard to say :( We'll try the very new vllm with Trinity and I'll report back. But anyway, good to have tests for this.

**Comment by vadimkantorov** [comment]
One thing which would be good is to explain clearer that if the prompt tokens already exceed min_tokens, then even an answer of one token would satisfy it.

I think this is not very obvious, as often we want to constraint output tokens, not input plus output
